packages/ui/src/components/Button/SButton.vue

```diff
@@ -0,0 +1,248 @@
+<script lang="ts">
+export default defineComponent({ name: 'SButton' })
+</script>
+
+<script setup lang="ts">
+import { computed, ComputedRef } from 'vue'
+import { SSpinner } from '@/lib'
+
+const BUTTON_TYPE_VALUES = ['primary', 'secondary', 'outline', 'action'] as const
+const BUTTON_SIZE_VALUES = ['mini', 'small', 'medium', 'big'] as const
+const BUTTON_ICON_POSITION_VALUES = ['left', 'right'] as const
+
+type ButtonType = typeof BUTTON_TYPE_VALUES[number]
+type ButtonSize = typeof BUTTON_SIZE_VALUES[number]
+type ButtonIconPosition = typeof BUTTON_ICON_POSITION_VALUES[number]
```

Comment:
Hm, I think it's better to move it from the component internals, to the `types.ts` file (it could be used for Storybook, for example)

Reply:
I think so too, but I can't decide should I move arrays to consts.ts or to types.ts. There are types using consts and then consts using types. I see to ways.
1. To create single file like `helpers.ts` or `api.ts` and just move this part there.
2. To replace strings in arrays with enum items and use enums instead of types. But the main idea of this arrays is the only one sourse of values.

Reply:
Anyway your solution is nice.